### PR TITLE
Fix Wildcard entries to return empty data instead of invalid path

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: gnmi_server/server_test.go [hunk 1]
```diff
@@ -3487,6 +3487,77 @@ func TestClientConnections(t *testing.T) {
 	}
 }
 
+func TestNonExistentTableNoError(t *testing.T) {
```
Comment (qiluo-msft) [inline gnmi_server/server_test.go:3490]: >TestNonExistentTableNoError [](http://example.com/codeflow?start=5&length=27)

Is there a test case for existing table no error?

Comment (zbud-msft) [reply gnmi_server/server_test.go:3582]: Added test

**Review by qiluo-msft — APPROVED** [review]

**Review by xincunli-sonic — APPROVED** [review]

**Review by ganglyu — APPROVED** [review]

**Review by FengPan-Frank — APPROVED** [review]

File: gnmi_server/server_test.go [hunk 2]
```diff
@@ -3487,6 +3487,169 @@ func TestClientConnections(t *testing.T) {
...
+
+			mutexNoti.Lock()
+
+			if len(gotNoti) == 0 {
+				t.Errorf("expected non zero notifications")
+			}
+
+			if diff := pretty.Compare(tt.wantNoti, gotNoti); diff != "" {
+				t.Log("\n Want: \n", tt.wantNoti)
+				t.Log("\n Got: \n", gotNoti)
```
Comment (qiluo-msft) [inline gnmi_server/server_test.go:3571]: We also need to verify if the gnmi response is correct. 
Seems like a bug in your manual test: "{}"

Comment (zbud-msft) [reply gnmi_server/server_test.go:3571]: Added test to verify gnmi response is empty json, please take a look at EMPTY_JSON.txt.

It is not a bug, as gnmi_client as part of subscribe will print out json_ietf_value as json string and "{}" is a valid response.

json_ietf structured value should be a valid json string, which empty json represented as string, "{}", is. 
https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#23-structured-data-types

If we use gnmi_get which prints out json and not json string, we indeed get expected empty json
 ------------------------- 
 [elem {
  name: "PSU_INFO"
}
]
The GetResponse is below
-------------------------

{}
-------------------------

**Comment by gechiang** [comment]
@qiluo-msft , @zbud-msft has made the changes per your review comment.  if the new changes looks good and no other comments, can you you help approve/merge this PR?
Thanks!

**Comment by gechiang** [comment]
@BYGX-wcr , please help monitor this PR state.  When it is merged, please work with @zbud-msft to port this needed fix to internal 202205 chassis branch via patch.  Thanks!

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**